src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolutionPriorityTests.vb

```diff
@@ -1560,5 +1562,1613 @@ End Module
+{
+    public static void M(this I2 x) => throw null;
+}
+" + OverloadResolutionPriorityAttributeDefinitionCS, parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source = "
+public class Program 
+    Shared Sub Main
+        Dim i3 As I3 = Nothing
+        i3.M()
+    End Sub
+End Class
+"
+
+            Dim compilation = CreateCompilation(source, references:={reference}, options:=TestOptions.DebugExe)
+            compilation.AssertTheseDiagnostics(
+<expected>
+BC30521: Overload resolution failed because no accessible 'M' is most specific for these arguments:
+    Extension method 'Public Sub M()' defined in 'C1': Not most specific.
+    Extension method 'Public Sub M()' defined in 'C2': Not most specific.
+        i3.M()
+           ~
+</expected>)
+        End Sub
+
+        <Fact>
+        Public Sub Overrides_NoPriorityChangeFromBase_Methods_CS()
+
+            Dim reference = CreateCSharpCompilation("
+using System.Runtime.CompilerServices;
+
+public class Base
+{
+    [OverloadResolutionPriority(1)]
+    public virtual void M(object o) => throw null;
+    public virtual void M(string s) => throw null;
+}
+
+public class Derived : Base
+{
+    public override void M(object o) => System.Console.WriteLine(""1"");
+    public override void M(string s) => throw null;
+}
+" + OverloadResolutionPriorityAttributeDefinitionCS, parseOptions:=New CSharpParseOptions(CSharp.LanguageVersion.Latest)).EmitToImageReference()
+
+            Dim source = "
+public class Program 
+    Shared Sub Main
+        Dim d = new Derived()
+        d.M(""test"")
+    End Sub
+End Class
+"
+
+            Dim compilation = CreateCompilation(source, references:={reference}, options:=TestOptions.DebugExe)
+            CompileAndVerify(compilation, expectedOutput:="1").VerifyDiagnostics()
+        End Sub
+
+        <Fact>
+        Public Sub Overrides_ChangePriorityInSource_Methods()
+
+            Dim source = "
+Imports System.Runtime.CompilerServices
+
+public class Base
+    <OverloadResolutionPriority(1)>
+    public Overridable Sub M(o As Object)
+        throw DirectCast(Nothing, System.Exception)
+    End Sub
+    public Overridable Sub M(s As String)
+        throw DirectCast(Nothing, System.Exception)
+    End Sub
+End Class
+
+public class Derived
+    Inherits Base
+
+    <OverloadResolutionPriority(0)>
+    public Overrides Sub M(o As Object)
+        System.Console.WriteLine(""1"")
+    End Sub
+    <OverloadResolutionPriority(2)>
+    public Overrides Sub M(s As String)
+        throw DirectCast(Nothing, System.Exception)
+    End Sub
+End Class
+
+public class Program 
+    Shared Sub Main
+        Dim d = new Derived()
+        d.M(""test"")
+    End Sub
+End Class
+"
+
+            Dim compilation = CreateCompilation({source, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.DebugExe)
+
+            ' Note, VB silently ignores attributes at wrong location 
```

Comment:
>Note, VB silently ignores attributes at wrong location [](http://example.com/codeflow?start=14&length=54)

Do we need a test for C# consuming such cases?

Reply:
It looks like there is such a test already: `Overrides_ChangePriorityInMetadata_Methods`.